### PR TITLE
Remove Renderer3 and renderComponent

### PR DESCRIPTION
**Pull request by alxhub**
See individual commits

### CODE REVIEW

**Comment by alfaproject** [comment]
OOC was it for performance reasons you decided to drop renderer 3, or? What was the goal/experiment in the first place? I've tried to search for my own answer but couldn't find, sorry if it's written somewhere already.

**Comment by alxhub** [comment]
@alfaproject Angular has historically had a "Renderer" abstraction (currently, the interface `Renderer2`). This adds overhead over direct DOM manipulation, but gives a number of benefits (it's a natural layer, for example, to implement CSS encapsulation and animation of added/removed elements). Theoretically, an application could swap out the underlying DOM entirely, and use the framework to render in a completely different context (like NativeScript for example).

`Renderer3` was a somewhat successful experiment to remove the overhead of the renderer abstraction by using the DOM itself directly. While this did work and was faster, it had several challenges:

1. As mentioned above, the renderer layer is used for things like CSS encapsulation and animations. It was unclear how users who needed to directly manipulate DOM should do so in a way that was compatible with these features, without the Renderer abstraction.

2. Having both codepaths (the production Renderer2 path, and the experimental path) in the runtime adds non-tree-shakable code to support the unused experiment, as well as complexity that anyone working on runtime code would need to navigate.

3. Many of our tests were using the experimental rendering approach, which meant they were not testing "reality" - the test configuration looked different than that of a real application. This could lead to bugs.

4. Even though the direct-to-DOM rendering system was intended to be only an internal experiment, the way it was implemented meant that production applications could accidentally instantiate a component in such a way as to activated. There was at least one case of this happening within Google, causing some very strange errors that were very difficult for the engineer who stumbled over this to debug.

So we're removing this experiment, because we've learned everything we needed to from it, we don't see a way to easily ship it, and it's costing us too much to keep the code supporting it in place.

File: packages/core/src/render3/instructions/styling.ts [hunk 1]
```diff
@@ -786,7 +787,8 @@ function findStylingValue(
       valueAtLViewIndex = isStylingMap ? EMPTY_ARRAY : undefined;
     }
     let currentValue = isStylingMap ? keyValueArrayGet(valueAtLViewIndex, prop) :
-                                      key === prop ? valueAtLViewIndex : undefined;
+        key === prop                ? valueAtLViewIndex :
```
Comment (AndrewKushnir) [inline packages/core/src/render3/instructions/styling.ts:790]: The formatting looks very unusual... May be wrap internal expression into `(...)`?

**Review by AndrewKushnir — APPROVED** [review]
@alxhub looks great 👍

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

File: packages/core/src/render3/instructions/styling.ts (hunk 1 above)
Comment (alxhub) [reply packages/core/src/render3/instructions/styling.ts:790]: Lol, no kidding. The formatter really gave up here.

**Review by atscott — APPROVED** [review]
reviewed-for: size-tracking, public-api

**Review by pkozlowski-opensource — APPROVED** [review]
Yay! 🎉

**Comment by alfaproject** [comment]
@alxhub thank you for the detailed answer. Today I learned something new (:

**Comment by atscott** [comment]
This PR was merged into the repository by commit 8d2e5e6c4f5dd8ff435152f4a3ce94d7197255aa.

**Pull request closed without merging**